src/main/java/AppRunner.java

```diff
@@ -13,13 +23,13 @@ public static void main(String[] args) {
         String mode = args[0];
         if (mode.equals("server")) {
             try {
-                AcceptingSocket listeningSocket = new ListeningServerSocket(new ServerSocket(portNumber));
+                AcceptingSocket listeningSocket = new ListeningSocket(new ServerSocket(portNumber));
                 EchoServer echoServer = new EchoServer(listeningSocket);
                 CommunicatingServer communicatingServer = new CommunicatingServer(echoServer.listenForConnection(), printer);
 
                 communicatingServer.run();
             } catch (IOException e) {
-                e.printStackTrace();
+                throw new ConnectionException(e.getMessage());
```

Comment:
Interesting tidbit: its a good practice to add the exception that happened with the new one you are throwing.
While it might sound odd, it's important for you as a developer when you see the exception that it retains as much _context_ as possible.

I'll leave a comment in the exception where to add it in :)